src/transactions/SponsorshipUtils.cpp

```diff
@@ -681,11 +721,33 @@ removeEntryWithPossibleSponsorship(AbstractLedgerTxn& ltx,
 {
     if (le.ext.v() == 1 && le.ext.v1().sponsoringID)
     {
-        auto sponsoringAcc = loadAccount(ltx, *le.ext.v1().sponsoringID);
+        // claimable balances are not subentries, so there's no sponsored
+        // account
+        LedgerEntry* sponsoredAccount =
+            le.data.type() == CLAIMABLE_BALANCE ? nullptr : &acc.current();
 
-        canRemoveEntryWithSponsorship(header.current(), le,
-                                      sponsoringAcc.current(), &acc.current());
-        removeEntryWithSponsorship(le, sponsoringAcc.current(), &acc.current());
+        if (acc.current().data.account().accountID.ed25519() ==
```

Comment:
what's with accessing `ed25519` directly? We should just compare `AccountID`

Reply:
Fixed